### PR TITLE
Support for declarative Response Handler chaining for Command Handlers

### PR DESCRIPTION
**Pull request by kipz**

### CODE REVIEW

File: operations/HandlerChains.ts [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+        this.success = builder;
+        return this;
+    }
+    public onError(builder: HandlerChain): this {
+        this.error = builder;
+        return this;
+    }
+    public build(): HandlerChainDescriptor {
+        const chain: HandlerChainDescriptor = { handle: this.handle };
+        if (this.any !== undefined) {
```
Comment (ddgenome) [inline operations/HandlerChains.ts:41]: Is this guaranteed to be `undefined` and not `null`?

File: operations/HandlerChains.ts [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+        }
+
+        this.walk(chain.onSuccess, nextOnSuccess, undefined);
+        this.walk(chain.onError, undefined, nextOnError);
+        this.walk(chain.onAny, nextOnSuccess, nextOnError);
+    }
+}
+
+/**
+ * Walk the chain, creating and linking all the response handlers
```
Comment (ddgenome) [inline operations/HandlerChains.ts:172]: Does this comment belong here?

**Review by ddgenome — COMMENTED** [review]
Even reading the tests it is not clear to mean what all these different but similarly named classes mean and how to use them. Perhaps some information could be added as TypeDoc and/or instructions for use could be added to the README?

File: operations/HandlerChains.ts (hunk 1 above)
Comment (kipz) [reply operations/HandlerChains.ts:41]: no. People can always explicitly set things to null, or other values that don't work, and we don't guard against this. They will get errors if they do however. Do you think we need to?

**Comment by kipz** [comment]
I'll add some docs and resubmit. Thanks for the review.

**Pull request closed without merging**